bin/plugin/commands/release.js

```diff
@@ -67,31 +68,16 @@ async function runReleaseBranchCreationStep(
 	await runStep( 'Creating the release branch', abortMessage, async () => {
 		const packageJsonPath = gitWorkingDirectoryPath + '/package.json';
 		const packageJson = readJSONFile( packageJsonPath );
-		const parsedVersion = semver.parse( packageJson.version );
-
-		// Follow the WordPress version guidelines to compute the version to be used
-		// By default, increase the "minor" number but if we reach 9, bump to the next major.
-		if ( parsedVersion.minor === 9 ) {
-			version = parsedVersion.major + 1 + '.0.0-rc.1';
-			releaseBranch = 'release/' + ( parsedVersion.major + 1 ) + '.0';
-			versionLabel = parsedVersion.major + 1 + '.0.0 RC1';
-		} else {
-			version =
-				parsedVersion.major +
-				'.' +
-				( parsedVersion.minor + 1 ) +
-				'.0-rc.1';
-			releaseBranch =
-				'release/' +
-				parsedVersion.major +
-				'.' +
-				( parsedVersion.minor + 1 );
-			versionLabel =
-				parsedVersion.major +
-				'.' +
-				( parsedVersion.minor + 1 ) +
-				'.0 RC1';
-		}
+		const nextMajor = getNextMajorVersion( packageJson.version );
+		const parsedMajorVersion = semver.parse( nextMajor );
+
+		version = nextMajor + '-rc.1';
+		releaseBranch =
+			'release/' +
+			parsedMajorVersion.major +
+			'.' +
+			parsedMajorVersion.minor;
```

Comment:
It's a bit odd that (via `getNextMajorVersion`) we deconstruct the version to determine the next version, only to then re-deconstruct it to generate the release branch.

Makes me wonder if `getNextMajorVersion` should return in a different form to be more reusable (e.g. array of parts, object of `major`, `minor`), or at least if it could be implemented as a separate function which is utilized by both this and the "higher-level" `getNextMajorVersion`.

Reply:
True but for me an API shouldn't necessarily be modeled for a given use-case. We could potentially add `getReleaseBranch( version )` if it's used more than once.